Digital io (#7)
hostmobility/documentation: docs/interfaces/mxv/digital_io.md

---
title:  MX-V Digital I/O
tags:
    - MX-V
---

## Overview

> MX-V has 8 digital out where it can both be used as a sink or source on the same pin called either digital-out-source or digital-out-sink with a number.

Reviewer: ```suggestion
MX-V has 8 digital out that can be used as either sink or source on the same pin: digital-out-source or digital-out-sink with a number.
```

 gpio-510 (digital-out-sink-1  )
 gpio-511 (digital-out-enable  )
```

## Digital in
This unit has 13 digital and analog in (adc) channels using iio devices.

```bash
 gpioget $(gpiofind digital-in-1)
 gpioget $(gpiofind digital-in-2)
 gpioget $(gpiofind digital-in-3)
 gpioget $(gpiofind digital-in-4)
 gpioget $(gpiofind digital-in-5)
 gpioget $(gpiofind digital-in-6)
 gpioget $(gpiofind digital-in-7)
 gpioget $(gpiofind digital-in-8)

 # analog way to access the value
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_0_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_1_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_2_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_3_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_4_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_5_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_6_raw
 cat /sys/bus/iio/devices/iio:device0/*DIG_IN_AN_7_raw

 cat /sys/bus/iio/devices/iio:device0/*AN_IN_AN_1_raw
 cat /sys/bus/iio/devices/iio:device0/*AN_IN_AN_2_raw
 cat /sys/bus/iio/devices/iio:device0/*AN_IN_AN_3_raw
 cat /sys/bus/iio/devices/iio:device0/*AN_IN_AN_4_raw
 cat /sys/bus/iio/devices/iio:device0/*AN_IN_AN_5_raw
```

> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
